ladybug/wea.py

```diff
@@ -547,60 +643,113 @@ def global_horizontal_irradiance(self):
                                 self.diffuse_horizontal_irradiance):
             sun = sp.calculate_sun_from_date_time(dt)
             glob_horiz.append(dhr + dnr * math.sin(math.radians(sun.altitude)))
-        return HourlyContinuousCollection(header_ghr, glob_horiz)
+        return self._aligned_collection(header_ghr, glob_horiz)
 
     @property
     def direct_horizontal_irradiance(self):
-        """Returns the direct irradiance on a horizontal surface at each timestep.
+        """Get a data collection for the direct irradiance on a horizontal surface.
 
         Note that this is different from the direct_normal_irradiance needed
-        to construct a Wea, which is NORMAL and not HORIZONTAL."""
-        analysis_period = AnalysisPeriod(timestep=self.timestep,
-                                         is_leap_year=self.is_leap_year)
+        to construct a Wea, which is NORMAL and not HORIZONTAL.
+        """
         header_dhr = Header(data_type=DirectHorizontalIrradiance(),
                             unit='W/m2',
-                            analysis_period=analysis_period,
+                            analysis_period=self.analysis_period,
                             metadata=self.metadata)
         direct_horiz = []
         sp = Sunpath.from_location(self.location)
         sp.is_leap_year = self.is_leap_year
         for dt, dnr in zip(self.datetimes, self.direct_normal_irradiance):
             sun = sp.calculate_sun_from_date_time(dt)
             direct_horiz.append(dnr * math.sin(math.radians(sun.altitude)))
-        return HourlyContinuousCollection(header_dhr, direct_horiz)
+        return self._aligned_collection(header_dhr, direct_horiz)
 
-    @property
-    def is_leap_year(self):
-        """Return the timestep."""
-        return self._is_leap_year
+    def filter_by_pattern(self, pattern):
+        """Filter this Wea based on a list of booleans.
```

Comment:
Since we are returning a new Wea object I suggest to change the docstring to reflect that:

```suggestion
        """Create a new filtered Wea from this Wea based on a list of booleans.
```

Reply:
Similar comment about the next couple of methods.